ui/shared/components/panel/variants/VariantClassify.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Button, Label } from 'semantic-ui-react'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+import { HttpRequestHelper } from 'shared/utils/httpRequestHelper'
+import { ColoredIcon } from 'shared/components/StyledComponents'
+import AcmgModal from '../acmg/AcmgModal'
+import { VerticalSpacer } from '../../Spacers'
+import PopupWithModal from '../../PopupWithModal'
+import { getVariantMainTranscript } from '../../../utils/constants'
+
+const CLINGEN_ALLELE_REGISTRY_URL = 'http://reg.genome.network/allele'
+const CLINGEN_VCI_URL = 'https://curation.clinicalgenome.org/select-variant'
+
+class ClinGenAlleleId extends React.PureComponent {
+
+  static propTypes = {
+    hgvsc: PropTypes.string,
+  }
+
+  state = {
+    copied: false,
+    alleleId: null,
+  };
+
+  constructor(props) {
+    super()
+
+    const { hgvsc } = props
+    new HttpRequestHelper(CLINGEN_ALLELE_REGISTRY_URL,
```

Comment:
you need to keep track of loading and show a spinner while the ID is fetching. Even if this happens very quickly for you locally, there might be a day when the allele registry is laggy. It probably best to use the `DataLoader` component 

```

load = (hgvsc) => {
  this.setState({ loading: true })
  new HttpRequestHelper(CLINGEN_ALLELE_REGISTRY_URL,
    ...
}

render() {
  const { hgvsc } = props
  const { alleleId, loading } = this.state
  return (
    <DataLoader contentId={hgvsc} content={alleleId} loading={loading} load={this.load}>
      ...
    </DataLoader>
  )
}

```